### PR TITLE
Fix concat-satisfy on GHC 9.2.1

### PR DESCRIPTION
**Pull request by sellout**
This broadens the CI coverage and fixes an issue that was missed by
testing 9.2.4 instead of 9.2.1.

### CODE REVIEW

**Comment by sellout** [comment]
Unfortunately, 83f06ae9e0c2629dde505ff678e186586a905a66 seems to have broken `concat-plugin` on everything before GHC 9.4. I think it was probably difficult to notice because there were already a bunch of broken checks due to gold tests, etc. So I added branch protection to master so that the checks passing (as of this PR) are marked required, and we can hopefully tighten that going forward.

I don't use concat-plugin, so the remaining breakage doesn't affect me, but the fixes here address failures in `concat-satisfy` and `concat-classes` that do.

**Comment by mikesperber** [comment]
Thanks, @sellout. I was going to investigate the CI failures (plugin works for me, as do the gold tests), but ran out of time. I'll get back to it  next week.

Meanwhile, my experience has been that there's no version jump in ghc that doesn't hold the risk of breaking changes. So I suggest also adding the latest known version along each 9.x line at least.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,11 +11,13 @@ jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
         ghc:
-          - "9.0.2"
-          - "9.2.4"
-          - "9.4.8"
+          - "9.0.1"
```
Comment (mikesperber) [inline .github/workflows/ci.yml:17]: I think we should also do the last version of each branch, but we can do that separately.

Comment (sellout) [reply .github/workflows/ci.yml:17]: We get the latest release covered in the Nix builds. Nix tends to not keep all the releases around, so I test the newer ones via Nix and the oldest via Cabal directly.

It does seem like the gold tests are passing in 9.4.8 (on Linux), but not in 9.4.1.

**Pull request merged**